### PR TITLE
0.9 supported

### PR DESCRIPTION
**Pull request by jacekgrzybowski**
Hello,

This pull request is meant for 0.9.x support. Queries for shards and series are yet to be implemented, besides that it's fully funtional (and covered with tests). I merged the latest changes (#88 and others) into my branch.

### CODE REVIEW

**Comment by toddboom** [comment]
@jacekgrzybowski this is fantastic - thank you! it looks like `jruby-head` is still broken on CI, but everything else is green, so i'll merge this in.

**Comment by cosmo0920** [comment]
:+1:. Super cool!

**Comment by jacekgrzybowski** [comment]
@toddboom Great! I will merge the latest changes (including #90) into my branch tomorrow morning.

**Comment by toddboom** [comment]
@jacekgrzybowski awesome - i can do it also, but if you can do that i'll merge it in this evening and cut `v0.2.0` in the morning!

**Comment by recurser** [comment]
I've added a [small PR](https://github.com/efigence/influxdb-ruby/pull/1) to the PR - I wasn't sure whether to make a separate PR here or not. Happy to re-submit here if that's preferable.

Awesome work @jacekgrzybowski btw!

File: spec/influxdb/cases/query_cluster_spec.rb [hunk 1]
```diff
@@ -17,57 +17,49 @@
...
 
-  describe "POST #update_cluster_admin" do
-    it "updates a cluster admin" do
-      stub_request(:post, "http://influxdb.test:9999/cluster_admins/adminadmin").with(
-        query: { u: "username", p: "password" },
-        body: { password: "passpass" }
-      )
+  describe "#list_cluster_admins" do
+    let(:response) { {"results"=>[{"series"=>[{"columns"=>["user", "admin"], "values"=>[["dbadmin", true], ["foobar", false]]}]}]} }
+    let(:expected_result) { [{"username"=>"dbadmin"}] }
```
Comment (palkan) [inline spec/influxdb/cases/query_cluster_spec.rb]: Maybe it should return a list of usernames, e.g. ; `let(:expected_result) { ["dbadmin"] }`.

**Comment by palkan** [comment]
Great work!

But...
Did you noticed `.rubocop.yml`? What about code style?)

Too much code is commented out. I think, there is a lot of functionality to be implemented. It would be great to add a checklist to the PR.

File: spec/influxdb/cases/query_cluster_spec.rb (hunk 1 above)
Comment (jacekgrzybowski) [inline spec/influxdb/cases/query_cluster_spec.rb]: Well, you might be right. I tried to keep it compatible with old API at first but it changed so much anyway that returning an array here can be a good idea.

**Comment by jacekgrzybowski** [comment]
@recurser thanks! I'm going to merge your changes into this PR.

@palkan I'm aware that you introduced rubocop in #88, but I had started working on 0.9.x lib before your changes were merged into master. Most errors yielded by rubocop were complaining about using old hash syntax and I have just converted all hashes to use ruby 1.9 syntax. However there are still some errors, mostly concerning missing spaces inside hashes. Fixing all of them manually would take a lot of time, and despite the fact that it should be done I don't think it's as crucial as getting this lib to support 0.9.x as soon as possible.

About commented out code - I'm going to implement `series` methods today. Only `shard` and `shard_space` methods won't be supported. If you're willing to search through the new docs and find a way to support them - feel free to contribute.

**Comment by palkan** [comment]
@jacekgrzybowski It's much easier to restyle code than you think) I'll made a PR.

**Comment by jacekgrzybowski** [comment]
@palkan Well yeah I see you used `rubocop` for that - I merged your changes, thanks.

**Comment by jacekgrzybowski** [comment]
Concerning missing methods mentioned by @palkan -  [schema exploration docs](https://influxdb.com/docs/v0.9/query_language/schema_exploration.html) describe sql-like keywords (`WHERE`, `FROM`, `WITH`) that you can apply to queries such as `SHOW SERIES`, `SHOW MEASUREMENTS` and so on. Because of that, I'm not sure if client should get separate methods for it, which could limit possibilities to build advanced queries (unless you build advanced querying interface like `activerecord`). As for now, it's possible get same results using `query` method: `client.query("SHOW SERIES FROM cpu_load WHERE region = 'uswest'")` and I'm leaving it as it is to let maintainers decide what to do next. I also haven't found anything about `shards` nor `shards space` in [query language spec](https://influxdb.com/docs/v0.9/query_language/spec.html).

**Comment by jacekgrzybowski** [comment]
@toddboom Yesterday I merged the lastest changes from master, added some more methods and merged PRs from @palkan and @recurser. I think it's ready to be merged in.

**Comment by palkan** [comment]
@jacekgrzybowski I think we should choose between two options: either we add `SHOW SMTH` shortcuts for all kinds of objects (series, measurements, tags, users, retention policies), either we don't add anything that can be done with `client.query('show ...')`. What do you think?

**Pull request merged**